### ISSUE TITLE
Keep Only The Last Two Eras In Consensus

### ISSUE THREAD

**Issue by xcthulhu** [casper-network/casper-node#1915]

**Comment by afck** [comment]
We should disentangle these questions:

#### How many earlier switch blocks do you need to initialize an era?

To initialize era `n`, you need at least:
* The era's booking block, which is in era `n - auction_delay - 1`.
* The era's key block, from era `n - 1`: for the random seed, and so that we can exclude inactive validators from becoming leaders.
* All blocks in between those, so we can ban all validators that equivocated in the meantime.

#### Across how many eras can you cite evidence?

This is currently the maximum: You can put an accusation into a new block in era `N`, if any bonded era (i.e. no older than `unbonding_delay - auction_delay`) contains evidence.

We are going to reduce this value to 1, assuming that in practice an equivocation in an era older than the previous one will have no effect anyway.

#### How many past eras does the era supervisor need to keep in memory?

Based on the previous point, we won't be able to punish validators who equivocate in any era earlier than the previous one anymore anyway, so for equivocation detection there's no point in keeping more than that.
However, with each _full_ era (including units) we keep, we also need to keep enough information — the validator set and who was banned and excluded from the leader sequence — about the preceding era to validate evidence, since that could be cited by units from the full era.

The other reason to keep old eras around is to help other nodes to catch up; with [CEP 36](https://github.com/casper-network/ceps/blob/master/text/0036-consensus-memory-usage.md) this won't be necessary anymore, and it's already questionable whether we need to help nodes that fell behind by more than one era: they should restart instead, and sync again.

#### In summary:

While in era `N`:
* The era supervisor will fully keep eras `N` and `N - 1` in memory.
* It keeps basic information about era `N - 2` for validating evidence.
* For initialization it needs all switch blocks starting at era `N - 3 - auction_delay`.

**Comment by piotr-dziubecki** [comment]
@goral09 @afck - is this already addressed?

**Comment by afck** [comment]
No, not done yet.

**Comment by afck** [comment]
Note that this has only been merged to the fast sync feature branch, not to `dev` yet.
